streams/src/main/java/org/apache/kafka/streams/kstream/type/TypeWithTypeArgs.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.type;
+
+import java.util.Arrays;
+
+class TypeWithTypeArgs implements java.lang.reflect.Type {
```

Comment:
No. ParameterizedType is an interface, and its method are not appropriate for our use.
We want a type representation that every type variable is resolved. It is better to have our own class to represent such a type.